onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -1148,6 +1152,39 @@ of [N, 0] then [N, 0].
         updateOutputShape(ctx, 0, output_shape);
       });
 
+  ONNX_CONTRIB_OPERATOR_SCHEMA(MatMulInteger16)
```

Comment:
What is the destiny of this op?
Do you plan to integrate this into MatmulInteger eventually, updgrade this as a separate onnx op or always keep this as contrib op?

Reply:
I heard there are some push-backs on int16 in MatMulInteger in onnx spec, and that's why this is added as a contrib op in MS domain. This can be easily changed once onnx spec supports this functionality, either in a way of extending types in MatMulInteger, or by adding a new MatMulInteger16 op.

---
In reply to: [303682422](https://github.com/microsoft/onnxruntime/pull/1391#discussion_r303682422) [](ancestors = 303682422)